packages/mui-system/src/ThemeProvider/ThemeProvider.test.js

```diff
@@ -16,7 +16,9 @@ describe('ThemeProvider', () => {
     let engineTheme;
 
     function Test() {
+      // TODO: uncomment once we enable eslint-plugin-react-compiler // eslint-disable-next-line react-compiler/react-compiler -- privateTheme is required outside the component
```

Comment:
You need to add this comment to the rest of tests in this file that do the same reassignments.

Reply:
sorry missed those... added comments in this commit -> https://github.com/mui/material-ui/pull/42637/commits/b715f304ffb67cb13970b7d1df1f2679b74619e3